### PR TITLE
Create a FuseBox class to group multiple breakers

### PR DESCRIPTION
**Pull request by pranavrc**
Use case: To create multiple circuit breakers with different settings and group them into one 'fusebox' so they can all be tripped if one breaks.

### CODE REVIEW

**Review by stjohnjohnson — CHANGES_REQUESTED** [review]
This is a breaking change.  Please bump the major version.

**Comment by yahoocla** [comment]
Thank you for submitting this pull request, however I do not see a valid CLA on file for you. Before we can merge this request please visit https://yahoocla.herokuapp.com/ and agree to the terms. Thanks! :smile:

**Comment by FenrirUnbound** [comment]
I can't think of a scenario where one would need to trip other circuit breakers if a different one is tripped. What case would you apply this usage for?

**Comment by pranavrc** [comment]
@FenrirUnbound We need to create circuit breakers with specific timeouts for each jenkins API endpoint but we'd have to bind them all into a single cluster so that if one breaks, we trip all endpoints across the same cluster.

**Comment by tkyi** [comment]
This repository is using semantic release, so you'll need to add `BREAKING CHANGES` in your commit message somewhere in order to bump major version. You can leave the package.json as it was before.
(http://docs.screwdriver.cd/about/contributing#submitting-pull-requests)

File: test/index.test.js
Comment (petey) [inline test/index.test.js]: `assert.isTrue()`

File: test/index.test.js
Comment (petey) [inline test/index.test.js]: `assert.isFalse()`

File: index.js
Comment (petey) [inline index.js]: missing jsdoc

File: index.js
Comment (petey) [inline index.js]: missing jsdoc and tests

File: index.js
Comment (petey) [inline index.js]: missing jsdoc

File: index.js
Comment (petey) [inline index.js]: missing jsdoc and tests

File: index.js
Comment (petey) [inline index.js]: This is giving me a bad code smell. I feel like this is tying into something that should be a private api. CircuitBreaker should probably expose the events of its implementation directly.

File: index.js
Comment (petey) [inline index.js]: missing `@constructor`

**Review by petey — CHANGES_REQUESTED** [review]

**Review by FenrirUnbound — APPROVED** [review]
👍 

Seems fine to me

htkzv0ciacez

File: index.js
Comment (stjohnjohnson) [inline index.js]: Why is there a `createFuse` and `addFuse`?

File: index.js
Comment (stjohnjohnson) [inline index.js:219]: Please update the `README` to describe the new features as well as how to integrate with the old features (since the interface changed).

**Review by stjohnjohnson — CHANGES_REQUESTED** [review]

**Review by petey — APPROVED** [review]
The electrical engineer in me says that the fusebox analogy doesn't really hold up very well, but the code seems sound.

File: README.md
Comment (stjohnjohnson) [inline README.md:138]: This is valid but insane.

**Review by stjohnjohnson — APPROVED** [review]

File: README.md
Comment (pranavrc) [reply README.md:138]: <insert link to wat.talk>

**Pull request merged**
